packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -208,21 +208,49 @@ class _PosixUtils extends OperatingSystemUtils {
 
   @override
   void zip(Directory data, File zipFile) {
-    _processUtils.runSync(
-      <String>['zip', '-r', '-q', zipFile.path, '.'],
-      workingDirectory: data.path,
-      throwOnError: true,
-    );
+    try {
+      _processUtils.runSync(
+        <String>['zip', '-r', '-q', zipFile.path, '.'],
+        workingDirectory: data.path,
+        throwOnError: true,
+      );
+    } on ArgumentError {
```

Comment:
is command not found the only situation we'd see an `ArgumentError` here?

Reply:
Technically we could get it by passing `null` too. If we did not have the ProcessManager wrapper, this would actually be a ProcessException